src/lm.jl

```diff
@@ -156,7 +156,30 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(3,1) * deviance(mm)/dof_residual(mm))'
+
+    function _local(x::StridedVector)
+        Ac_ldiv_B!(R,x)
+        residvar * (x⋅x)
+    end
+
+    interval = similar(prediction)
+    for i in 1:length(interval)
+        interval[i] = _local(newx[i,:])[1]
+    end
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval)
```

Comment:
Another reason for having the `sqrt` outside the loop is that if I want to create a `predint` instead (a prediction interval) i need to multiply by the predictive variance before taking the square root. I can put a conditional in the inner function as well, of course.